api/v1/controllers/botData.js

```diff
@@ -90,6 +92,37 @@ module.exports = {
     doPost(req, res, next, helper);
   },
 
+  /**
+   * POST /botData/upsert
+   *
+   * Check if botData exists if does not exist create a new botData and
+   * sends it back in the response; if does exist then update the value
+   *
+   * @param {IncomingMessage} req - The request object
+   * @param {ServerResponse} res - The response object
+   * @param {Function} next - The next middleware function in the stack
+   */
+  upsertBotData(req, res, next) {
+    const queryBody = req.swagger.params.queryBody.value;
+    BotData.findOne({
+      where: {
+        name: queryBody.name,
+        botId: queryBody.botId,
+        roomId: queryBody.roomId,
+      },
+    })
+    .then((bd) => {
```

Comment:
Do you want to do this in a scope in the botData model?

Reply:
I dont think it needs to? Do you think it would function better?

Reply:
It has got nothing to do with performance, just that converting it to a scope makes it more modular and that you are not writing any DB logic in the controller.

Reply:
ive made some changes --- is this what you mean?

Reply:
This is fine too. I was thinking we could do something like https://github.com/salesforce/refocus/blob/master/db/model/sample.js#L222. Maybe for another day? I have approved this PR anyways.